Jenkins.io - adding documentation guidelines to CONTRIBUTING.adoc

kmartens27 opened the pull request:
This pull request is a suggestion to add some guidelines to adding documentation.
These are things that I've been trying to incorporate as Docs officer and a documentation contributor.
I suggested:

- single sentence per line formatting
- Using present tense as much as possible/where applicable
- Ensure there is sample output
- Making sure to use inclusive naming

If there are other guidelines that can be added, I'm more than happy to update those as well. My goal is to have a bit more structure around the documentation submissions, keeping in mind the global reach that the Jenkins community has.

krisstern commented on CONTRIBUTING.adoc:
```diff
@@ -246,6 +246,26 @@ Be aware that some of this content might already be present in the Jenkins User
+This helps ensure readers can assess their work as they progress through the documentation.
+
+* Always use the correct naming for Jenkins temrinology.
+We have been working on updating the terminology throughout Jenkins documentation to align with the link:https://www.jenkins.io/sigs/advocacy-and-outreach/#inclusive-naming[inclusive naming initiative].
```
```suggestion
We have been working on updating the terminology throughout Jenkins documentation to align with the link:/sigs/advocacy-and-outreach/#inclusive-naming[inclusive naming initiative].

```

krisstern commented on CONTRIBUTING.adoc:
```diff
@@ -246,6 +246,27 @@ Be aware that some of this content might already be present in the Jenkins User
+* Always use the correct naming for Jenkins temrinology.
+We have been working on updating the terminology throughout Jenkins documentation to align with the link:/sigs/advocacy-and-outreach/#inclusive-naming[inclusive naming initiative].
+
+** If you find instances of existing documenation where this should be updated, submit a PR with the fix, and we can make sure that the documentation is corrected.
```
```suggestion
** If you find instances of existing documentation where this should be updated, submit a PR with the fix, and we can make sure that the documentation is corrected.
```

krisstern commented:
I have been thinking it may not be a bad idea to add some instructions about adding code snippets, especially regarding the need to annotate the code with the syntax
```shell
[source,bash]
----------
Code here
----------
```
where I have taken the liberty to use `bash` for an example, which should be changed according to the actual context. This is particular useful for the newly proposed adding `copy-to-clipboard` feature as discussed in #6260.

kmartens27 commented:
I agree with you on that @krisstern. In fact, the asciidoc reference guide has a great page on how that works and how to set it up: https://docs.asciidoctor.org/asciidoc/latest/verbatim/source-blocks/ 

This page includes various examples and methods of putting the code block together. I could see sharing the most basic example, like you have done or the ones on the asciidoc page, and then encouraging users to follow the asciidoc guide. Does that line up with what you were thinking, or is there anything else that could be added/changed to make it better?

kmartens27 commented:
Sure thing @krisstern! I appreciate the insight and willingness to add on. By all means add/submit whatever you would like, and if I have any questions or suggestions I'll share them here!

krisstern commented:
Awesome @kmartens27! I have just added a subsection entitled "Adding a code snippet" to elaborate more on the best practice to add source blocks to AsciiDocs. Please see if this is an acceptable addition or not.

krisstern reviewed (approved):
Looks good

krisstern commented on CONTRIBUTING.adoc:
```diff
@@ -375,6 +396,24 @@ Redirects that need a shorter link are created by convention in the `content/red
+----
+-----
+
+In this example bash would be the name of the language language used. So this second quantity should be replaced with the name of the language used in the code snippet accordingly.
```
```suggestion
In this example bash would be the name of the language used. So this second quantity should be replaced with the name of the language used in the code snippet accordingly.
```

zbynek commented on CONTRIBUTING.adoc:
```diff
@@ -246,6 +246,27 @@ Be aware that some of this content might already be present in the Jenkins User
+This helps ensure readers can assess their work as they progress through the documentation.
+
+* Always use the correct naming for Jenkins temrinology.
+We have been working on updating the terminology throughout Jenkins documentation to align with the link:/sigs/advocacy-and-outreach/#inclusive-naming[inclusive naming initiative].
```
Maybe link to https://www.jenkins.io/doc/book/glossary/ ?

daniel-beck commented on CONTRIBUTING.adoc:
```diff
@@ -246,6 +246,27 @@ Be aware that some of this content might already be present in the Jenkins User
 
 The documentation pages can use the same metadata (`title`, `description`, `opengraph:image`) as blog posts.
 
+* When creating or adding documentation, make sure that every numbered line contains a complete sentence, including an ending punctuation mark (`.`, `?`, `!`).
```
Unclear. What's a numbered line? Are there exceptions? What about line 123 ending without punctuation, or line 248 having no content?

kmartens27 replied:
Good point on this @daniel-beck, I've reworded this so that it is more clear and provides a reference for the style.

daniel-beck commented on CONTRIBUTING.adoc:
```diff
@@ -246,6 +246,27 @@ Be aware that some of this content might already be present in the Jenkins User
+* Be sure to provide sample output when providing commands or code.
+This helps ensure readers can assess their work as they progress through the documentation.
+
+* Always use the correct naming for Jenkins temrinology.
```
```suggestion
* Always use the correct naming for Jenkins terminology.
```

daniel-beck commented on CONTRIBUTING.adoc:
```diff
@@ -246,6 +246,27 @@ Be aware that some of this content might already be present in the Jenkins User
+
+* When creating documentation, avoid future tense as much as possible.
+** Users should be able to approach documentation as though they are performing the actions as they read.
+This makes the text more active and encourages more interaction with the documentation itself.
```
```suggestion
   This makes the text more active and encourages more interaction with the documentation itself.
```

Etc. -- for readability.

daniel-beck commented on CONTRIBUTING.adoc:
```diff
@@ -246,6 +246,27 @@ Be aware that some of this content might already be present in the Jenkins User
+In cases of code examples or output, this is not necessary and should be formatted normally.
+----
+
+* When creating documentation, avoid future tense as much as possible.
```
Why is this a list?

kmartens27 replied:
I've updated it to not be a list, that was more for me to separate topics, but it is not needed in the actual documentation. Thanks @daniel-beck!

daniel-beck commented on CONTRIBUTING.adoc:
```diff
@@ -246,6 +246,27 @@ Be aware that some of this content might already be present in the Jenkins User
+* Always use the correct naming for Jenkins temrinology.
+We have been working on updating the terminology throughout Jenkins documentation to align with the link:/sigs/advocacy-and-outreach/#inclusive-naming[inclusive naming initiative].
+
+** If you find instances of existing documentation where this should be updated, submit a PR with the fix, and we can make sure that the documentation is corrected.
```
Perhaps useful to mention necessary exceptions (e.g. internal class names retained for compatibility, UI labels that haven't been updated in the software, etc.), because we had braindead PRs "fixing" those occurrences before.

kmartens27 replied:
Agreed, thanks for calling this out @daniel-beck. I'll be sure to include that here.

kmartens27 commented:
Thank you so much for the suggestions and feedback @daniel-beck. I moved the text to be in its own section so that it does not take away from the existing info. I've also reworked the content so that it is more specific when needed, and included the necessary exceptions to the terminology piece with an admonition block to highlight it.

NotMyFault reviewed (approved):
Thanks Kevin, much appreciated!
